R/grouped_epi_archive.R

```diff
@@ -246,20 +252,10 @@ epix_slide.grouped_epi_archive <- function(x, f, ..., before, ref_time_values,
     ref_time_values <- sort(ref_time_values)
   }
 
-  # Validate and pre-process `before`:
-  if (missing(before)) {
-    cli_abort("`before` is required (and must be passed by name);
-                if you did not want to apply a sliding window but rather
-                to map `epix_as_of` and `f` across various `ref_time_values`,
-                pass a large `before` value (e.g., if time steps are days,
-                `before=365000`).")
+  if (!checkmate::test_scalar(before)) {
+    cli_abort("`before` is a required scalar value.")
   }
-  before <- vctrs::vec_cast(before, integer())
-  assert_int(before, lower = 0L, null.ok = FALSE, na.ok = FALSE)
-
-  # If a custom time step is specified, then redefine units
-
-  if (!missing(time_step)) before <- time_step(before)
+  validate_slide_window_arg(before, guess_time_type(x$private$ungrouped$DT$version))
```

Comment:
Since much of our API data has time_value and issue in the same data type, I'll just require them to be the same for now. We can add some relaxations for these constraints in another PR.